test/TreasureMarketplace.test.ts

```diff
@@ -43,11 +43,18 @@ describe('TreasureMarketplace', function () {
     erc1155 = await ERC1155Mintable.deploy()
     await erc1155.deployed();
 
-    const newOwner = deployer;
     const TreasureMarketplace = await ethers.getContractFactory('TreasureMarketplace')
-    marketplace = await TreasureMarketplace.deploy()
-    await marketplace.deployed();
-    await marketplace.initialize(100, feeRecipient, magicToken.address);
+    const marketplaceImpl = await TreasureMarketplace.deploy();
+
+    const TreasureMarketplaceAbi = (await artifacts.readArtifact('TreasureMarketplace')).abi;
+    const iface = new ethers.utils.Interface(TreasureMarketplaceAbi);
+    const data = iface.encodeFunctionData("initialize", [100, feeRecipient, magicToken.address]);
+
+    const EIP173Proxy = await ethers.getContractFactory('EIP173Proxy')
```

Comment:
Ideally you'd use the fixtures feature of the deploy plugin rather than mocking the proxy.

Reply:
That's what I did first but then I'd have to refactor tests to use hardhat impersonating for token transfers because I cannot change payment token address after deployment. So I decided to do it this way.

Reply:
It's not high priority imo so don't worry, for reference, it could be done parametrising the token address in the deployment script and having the mock token deployed on non live networks.

Reply:
I'll refactor this during more peaceful time.